components/mosart/src/riverroute/RtmMod.F90

```diff
@@ -1932,21 +1970,32 @@ subroutine Rtmini(rtm_active,flood_active)
 !#endif
 
 
- do nt = 1,nt_rtm
-      do nr = rtmCTL%begr,rtmCTL%endr
-      
-       call UpdateState_hillslope(nr,nt)
-       call UpdateState_subnetwork(nr,nt)   
-                                          
-        rtmCTL%volr(nr,nt) = (TRunoff%wt(nr,nt) + TRunoff%wr(nr,nt) + TRunoff%wh(nr,nt)*rtmCTL%area(nr)*TUnit%frac(nr))  ! times "TUnit%frac( nr )" or not ?
-        if (inundflag .and. nt == 1) then  
-           rtmCTL%volr(nr,nt) = rtmCTL%volr(nr,nt) + TRunoff%wf_ini( nr )
-        else        
-           call UpdateState_mainchannel(nr,nt)
-        endif
+    do nt = 1,nt_rtm
```

Comment:
I saw in another file you changed the iteration of nt from 1:nt_rtm to nt_liq:nt_ice.

Reply:
Good point. I've split this block into two blocks: one for nt_liq and nt_ice and another for nt_san and nt_mud.